cfi/derive/src/lib.rs

```diff
@@ -0,0 +1,79 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    lib.rs
+
+Abstract:
+
+    File contains CFI procedural macro.
+
+References:
+    https://tf-m-user-guide.trustedfirmware.org/design_docs/tfm_physical_attack_mitigation.html
```

Comment:
I do not see counter decrements in their suggestions. Not sure a counter increment/decrement wrapper achieves actual CFI monitoring? Entire calls can still be skipped afaict.

Reply:
Based on the discussion we had today, we'll embed either the inc/dec operation in the CFI decorated function and corresponding opposite operation in the wrapper function. This will come as a separate PR.